packages/next-swc/crates/next-core/src/bootstrap.rs

```diff
@@ -0,0 +1,255 @@
+        reference_type::EcmaScriptModulesReferenceSubType,
+        resolve::parse::RequestVc,
+        virtual_asset::VirtualAssetVc,
+    },
+    ecmascript::{
+        resolve::esm_resolve, utils::StringifyJs, EcmascriptInputTransform,
+        EcmascriptInputTransformsVc, EcmascriptModuleAssetType, EcmascriptModuleAssetVc,
+        InnerAssetsVc,
+    },
+};
+use turbo_tasks::Value;
+use turbo_tasks_fs::{File, FileSystemPathVc};
+
+use crate::asset_helpers::as_es_module_asset;
+
+#[turbo_tasks::function]
+pub async fn route_bootstrap(
+    asset: AssetVc,
+    context: AssetContextVc,
+    base_path: FileSystemPathVc,
+    bootstrap_asset: AssetVc,
+    config: BootstrapConfigVc,
+) -> Result<EcmascriptModuleAssetVc> {
+    let resolve_origin = if let Some(m) = EcmascriptModuleAssetVc::resolve_from(asset).await? {
+        m.as_resolve_origin()
+    } else {
+        bail!("asset does not represent an ecmascript module");
+    };
+
+    // TODO: this is where you'd switch the route kind to the one you need
+    let route_module_kind = "app-route";
+
+    let resolved_route_module_asset = esm_resolve(
+        resolve_origin,
+        RequestVc::parse_string(format!(
+            "next/dist/server/future/route-modules/{}/module",
+            route_module_kind
+        )),
+        Value::new(EcmaScriptModulesReferenceSubType::Undefined),
+        OptionIssueSourceVc::none(),
+        IssueSeverity::Error.cell(),
+    );
+    let route_module_asset = match &*resolved_route_module_asset.first_asset().await? {
+        Some(a) => *a,
+        None => bail!("could not find app asset"),
+    };
+
+    Ok(bootstrap(
+        asset,
+        context,
+        base_path,
+        bootstrap_asset,
+        InnerAssetsVc::cell(indexmap! {
+            "ROUTE_MODULE".to_string() => route_module_asset,
+        }),
+        config,
+    ))
+}
+
+#[turbo_tasks::value(transparent)]
+pub struct BootstrapConfig(IndexMap<String, String>);
+
+#[turbo_tasks::value_impl]
+impl BootstrapConfigVc {
+    #[turbo_tasks::function]
+    pub fn empty() -> Self {
+        BootstrapConfigVc::cell(IndexMap::new())
+    }
+}
+
+#[turbo_tasks::function]
+pub async fn bootstrap(
+    asset: AssetVc,
+    context: AssetContextVc,
+    base_path: FileSystemPathVc,
+    bootstrap_asset: AssetVc,
+    inner_assets: InnerAssetsVc,
+    config: BootstrapConfigVc,
+) -> Result<EcmascriptModuleAssetVc> {
+    let path = asset.ident().path().await?;
+    let path = base_path
+        .await?
+        .get_path_to(&path)
+        .context("asset is not in base_path")?;
+    let path = if let Some((name, ext)) = path.rsplit_once('.') {
+        if !ext.contains('/') {
+            name
+        } else {
+            path
+        }
+    } else {
+        path
+    };
+
+    let pathname = normalize_app_page_to_pathname(path);
+
+    let mut config = config.await?.clone_value();
+    config.insert("PAGE".to_string(), path.to_string());
```

Comment:
According to the typings this also has `NAME`

Reply:
That's only provided by the edge transition